From 40defd0c4d6a2a3f4b40842c3b8798755a5c1ef1 Mon Sep 17 00:00:00 2001
From: Ruediger Meier <ruediger.meier@ga-group.nl>
Date: Tue, 23 Jan 2018 16:59:28 +0100
Subject: [PATCH] rfkill: provide RFKILL_TYPE_FM if undefined

As discussed last year it's nice to be compatible to 2.6.32
https://www.spinics.net/lists/util-linux-ng/msg13963.html

BTW also re-define NUM_RFKILL_TYPES if needed, although we are
not really using it.

Signed-off-by: Ruediger Meier <ruediger.meier@ga-group.nl>
---
 sys-utils/rfkill.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git sys-utils/rfkill.c sys-utils/rfkill.c
index 75804ad41..031fe436f 100644
--- sys-utils/rfkill.c
+++ sys-utils/rfkill.c
@@ -45,7 +45,12 @@
  * year 2009 (2.6.33) or older.
  */
 #ifndef RFKILL_TYPE_NFC
+# ifndef RFKILL_TYPE_FM
+#  define RFKILL_TYPE_FM	RFKILL_TYPE_GPS + 1
+# endif
 # define RFKILL_TYPE_NFC	RFKILL_TYPE_FM + 1
+# undef NUM_RFKILL_TYPES
+# define NUM_RFKILL_TYPES	RFKILL_TYPE_NFC + 1
 #endif
 
 struct rfkill_type_str {
